server/recceiver/cfstore.py

```diff
@@ -69,15 +70,49 @@ def commit(self, transaction_record):
 
     def __commit__(self, TR):
         _log.debug("CF_COMMIT %s", TR.infos.items())
-        pvNames = [unicode(rname, "utf-8") for rid, (rname, rtype) in TR.addrec.iteritems()]
-        delrec = list(TR.delrec)
-        iocName = TR.src.port
-        hostName = TR.src.host
-        iocid = hostName + ":" + str(iocName)
-        owner = TR.infos.get('CF_USERNAME') or TR.infos.get('ENGINEER') or self.conf.get('username', 'cfstore')
+        """
+        a dictionary with a list of records with their associated property info  
+        pvInfo 
+        {rid: { "pvName":"recordName",
+                "infoProperties":{propName:value, ...}}}
+        """
+                
+        iocName = TR.infos.get('IOCNAME') or TR.src.port
+        hostName = TR.infos.get('HOSTNAME') or TR.src.host
+        owner = TR.infos.get('ENGINEER') or TR.infos.get('CF_USERNAME') or self.conf.get('username', 'cfstore')
         time = self.currentTime()
+        
+        pvInfo = {}
+        for rid, (rname, rtype) in TR.addrec.iteritems():
+            pvInfo[rid] = {"pvName":rname}            
+        for rid, (recinfos) in TR.recinfos.iteritems():
+            if "properties" in  recinfos:
+                if rid in pvInfo:
+                    recProperties = recinfos["properties"]
+                    properties = []
+                    for prop in recProperties.split(","):
+                        p = prop.strip().split("=")
+                        properties.append({u'name': p[0], u'owner': owner, u'value': p[1]})
+                    pvInfo[rid]["infoProperties"] = properties
+                else:
+                    _log.exception("could not find the associated record for properties")
```

Comment:
```.exception()``` is only useful when called when there is an active exception (eg. within a ```catch:``` block).  Otherwise just use ```.error()```.